pkg/occlient/occlient.go

```diff
@@ -949,6 +761,160 @@ func (c *Client) UpdateBuildConfig(buildConfigName string, projectName string, g
 	return nil
 }
 
+// PatchCurrentDC "patches" the current DeploymentConfig with a new one
+// however... we make sure that configurations such as:
+// - volumes
+// - environment variables
+// are correctly copied over / consistent without an issue.
+func (c *Client) PatchCurrentDC(name string, dc appsv1.DeploymentConfig) error {
+
+	// Retrieve the current DC
+	currentDC, err := c.GetDeploymentConfigFromName(name, c.namespace)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get DeploymentConfig %s", name)
+	}
+
+	// Find the container (don't want to use .Spec.Containers[0] in case the user has modified the DC...)
+	// in order to retrieve what the volumes are
+	foundCurrentDCContainer, err := findContainer(currentDC.Spec.Template.Spec.Containers, name)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to find current DeploymentConfig container %s", name)
+	}
+
+	// Append the existing VolumeMounts to the new DC. We use "range" and find the correct container rather than
+	// using .spec.Containers[0] *in case* the template ever changes and a new container has been added.
+	for index, container := range dc.Spec.Template.Spec.Containers {
+		// Find the container
+		if container.Name == name {
+			// Loop through all the volumes
+			for _, volume := range foundCurrentDCContainer.VolumeMounts {
+				// If it's the supervisord volume, ignore it.
+				if volume.Name == supervisordVolumeName {
+					continue
+				} else {
+					dc.Spec.Template.Spec.Containers[index].VolumeMounts = append(dc.Spec.Template.Spec.Containers[index].VolumeMounts, volume)
+				}
+			}
```

Comment:
Can we break from here? Outside the inner for but inside if and after the inner for?